src/server/qgshttprequesthandler.cpp

```diff
@@ -218,21 +218,10 @@ void QgsHttpRequestHandler::sendResponse()
   clearBody();
 }
 
-QByteArray QgsHttpRequestHandler::getResponse( const bool returnHeaders,
-    const bool returnBody )
+QPair<QByteArray, QByteArray> QgsHttpRequestHandler::getResponse()
 {
-  if ( ! returnHeaders )
-  {
-    return mResponseBody;
-  }
-  else if ( ! returnBody )
-  {
-    return mResponseHeader;
-  }
-  else
-  {
-    return mResponseHeader.append( mResponseBody );
-  }
+  QPair<QByteArray, QByteArray> response( mResponseHeader, mResponseBody );
+  return response;
```

Comment:
@m-kuhn I'm still working on a new PR based on this (bindings are now working) but I need some c++ help to make sure what I'm doing here is safe. 
I'm using a QPair of QByteStream to return a Python tuple in the bindings (tests are passing happily and the conversion is fine) but I'm not sure that returning a stack-allocated QPair is a good idea, even if the two items are instance members and hence are not deleted when the method returns. This is probably working because it is only used in the bindings and the converted type returns the mReponse* member variables that are not deallocated.

The basic idea of this PR is to return the headers and body as a tuple when the server handleRequest is invoked from Python, this simplifies the API a lot and reduces the code and methods needed.

Any suggestion?

In case you need it, you can see the latest (working) code here: https://github.com/elpaso/QGIS/tree/qgsserver-return-pair